LDAP/ldap2s3/ldap2s3.py

```diff
@@ -112,21 +126,89 @@ def normalize_pgp(self, pgplist):
 
         return ret
 
+    def validate_user(self, user_profile):
+        """
+        Validate a user profile against CIS profile
+        Returns True if the profile validate, False if not.
+        @user_profile: dict
+        """
+        schema_file = self.cis_config.schema
+        try:
+            with open(schema_file) as fd:
+                schema = json.load(fd)
+        except FileNotFoundError:
+            return False
+
+        try:
+            jsonschema.validate(user_profile, schema)
+            return True
+        except jsonschema.exceptions.ValidationError:
+            return False
+
     def user(self, entry):
         """
         Finds canonical LDAP data for that user
         """
-        user = DotDict(dict())
-        attrs = entry.get('raw_attributes')
-        user.mail = self.gfe(attrs, 'mail')
-        user.dn = self.gfe(entry, 'raw_dn')
-        if not user.mail or not user.dn:
-            logger.warning('Invalid user specification dn: {} mail: {}'.format(user.dn, user.mail))
-
-        user.sshPublicKey = self.normalize_ssh(attrs.get('sshPublicKey'))
-        user.pgpFingerprint = self.normalize_pgp(attrs.get('pgpFingerprint'))
+        user = DotDict(json.loads(self.userprofile_json))
 
-        return user
+        # Add LDAP attributes
+        attrs = entry.get('raw_attributes')
+        dn = self.gfe(entry, 'raw_dn')
+
+        # Insert LDAP email as primary email
+        user.primaryEmail = self.gfe(attrs, 'mail')
+        email = {
+                  'value': user.primaryEmail,
+                  'verified': True,
+                  'primary': True,
+                  'name': 'LDAP-imported Email'
+                }
+        user.emails.append(email)
+
+        if not user.primaryEmail or not dn:
+            logger.warning('Invalid user specification dn: {} mail: {}'.format(dn, user.primaryEmail))
+
+        # Terrible hack to emulate the LDAP user_id
+        # This NEEDS to match Auth0 LDAP user_ids
+        # XXX Replace this by opaque UUIDs someday, as well as in the Auth0 LDAP Connector
+        user.userName = "{}".format(user.primaryEmail.split('@')[0])
+        user.user_id = "{}|{}".format(self.cis_config.user_id_prefix, user.userName)
```

Comment:
@jdow here I'm constructing a `user_id` that is effectively:
`ad|Mozilla-LDAP{-Dev}|xxx` where `xxx` is the first part of the LDAP primary email (ie jdow@mozilla.com => jdow).

I'm assuming this will work exactly like the auth0 LDAP connector. If it does not, users won't have their attributes properly set, so it's important to double check

Reply:
The auth0 LDAP connector does not take the xxx from the e-mail address, but rather looks at the LDAP "uid" attribute. I would recommend doing the same.

Reply:
note sure why github didnt collapse this, but this is now done